.github/workflows/build-and-test.yml

```
@@ -75,15 +99,23 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           # Build the library, tests, and examples without running them to avoid recompilation in the run tests step
-          command: test
-          args: --all --release --no-run
+          command: build
+          args: --workspace --tests --examples --release
 
       - name: Run tests
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --all --release
 
+      - name: Run Rust examples
+        working-directory: examples
+        # run examples only on ubuntu for now
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          cargo read-manifest > ../example-manifest.json
+          cd ../
+          cat ./example-manifest.json | for entry in $(jq -r '.targets[].name'); do cargo run --example "$entry" --release || exit 1; done
```

Two things:

1. You can set which manifest is passed to cargo using `manifest-path`. E.g. `cargo read-manifest --manifest-path ./examples/Cargo.toml`.
2. `cargo run --example <name>` works inside the `examples` directory too (but I see that doing so does _not_ use the pre-built examples from the previous step, weird).

I think this can be simplified either way.

```suggestion
      - name: Run Rust examples
        # Run examples only on Ubuntu for now.
        if: matrix.os == 'ubuntu-latest'
        run: |
          cargo read-manifest --manifest-path ./examples/Cargo.toml | for entry in $(jq -r '.targets[].name'); do cargo run --example "$entry" --release || exit 1; done

```

---

Yeah, the reason for 2. seems to be that `target` is resolved from the current working directory. Maybe there is a way to work around it but I like the idea of running from where it is build, so Option 1 it is :)